configure.ac

```diff
@@ -350,7 +350,31 @@ AM_CONDITIONAL([ENABLE_GRPTEST_LDMS_TEST], [test "x$ENABLE_LDMS_TEST_FALSE" = "x
 dnl options for contributed samplers
 OPTION_DEFAULT_DISABLE([ipmireader], [ENABLE_IPMIREADER])
 OPTION_DEFAULT_DISABLE([tutorial-sampler],[ENABLE_TUTORIAL_SAMPLER])
-
+OPTION_DEFAULT_ENABLE([variorum_ldms],[ENABLE_VARIORUM_LDMS])                   
+
+dnl check for Variorum dependencies                                             
+if test "${VARIORUM_DIR+set}" == set; then                                      
```

Comment:
This is going to need to be overhauled. We should not pass paths to library dependencies in environment variables that autoconf doesn't know about. Instead, the autoconf way to handle this is to create a AC_ARG_WITH blocks. People would do "configure --with-jansson=/some/path --with-variorum=/some/other/path".

So here's what I think we need:

First, drop the "OPTION_DEFAULT_ENABLE([variorum_ldms],[ENABLE_VARIORUM_LDMS])"
Next, add an AC_ARG_WITH block for jansson (this checks for the external jansson library)
Then add an AC_ARG_WITH block for variorum (this checks for the external variorum library)
Finally, if you think you need it, add an AC_ARG_ENABLE block for the "variorum_sampler". This would
  just check if the the previously mentioned AC_ARG_WITH checks passed.

Take a look at the AC_ARG_WITH geopm (later in this file) as a basic example of what I am suggesting.

Also note that we shouldn't be setting these paths in the global CFLAGS/LDFLAGS/CPPFLAGS. Instead there should be individual versions of each of those that are used where needed. You can see hos the geopm example saves and restores those global values around the tests for the libraries and headers.